lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1753,32 +1792,19 @@ void TFDeabstraction::checkAndCanonicalizeAttributes() {
   // first SIL instruction that could not be folded.
   SmallVector<SymbolicValue, 32> results;
   constantEvaluator.computeConstantValues(valuesToCheck, results);
+  assert(valuesToCheck.size() == results.size() && "incorrect values returned");
 
   // Transform the returned information about constants into a map that we can
   // query.  The results list should correspond directly to the values we asked
   // about.
   DenseMap<SILValue, SymbolicValue> constants;
-
-  assert(valuesToCheck.size() == results.size() && "incorrect values returned");
-  for (unsigned i = 0, e = valuesToCheck.size(); i != e; ++i) {
-    // If this value is a non-constant, just ignore it.
-    if (!results[i].isConstant())
-      continue;
-
-    // Otherwise, add the information about it to our map.
+  for (unsigned i = 0, e = valuesToCheck.size(); i != e; ++i)
     constants.insert({valuesToCheck[i], results[i]});
-  }
-
 
   // Now that we've computed whether any of the operands are constants,
   // substitute them into the operations that we have, eliminating abstractions.
   // This makes it immediately obvious to partitioning what is and isn't a
   // constant.
-  //
-  // TODO: When something *isn't* a constant, we should diagnose why: is it a
-  // use of a non-constexpr function that caused the problem, or the use of
-  // some other thing that we can't analyze, like passing through a class?
-  //
   for (auto *&inst : tensorOps) {
```

Comment:
Are we rewriting this pointer in `tensorOps` somewhere in the loop? Otherwise why use a reference?